### ISSUE TITLE
not able to access pdb4amber

### ISSUE THREAD

**Issue by DDGmichigan** [Amber-MD/pdb4amber#78]
Hi,
As i was reading through the Amber18 manual, it says we can type pdb4amber and it should prompt me the usage details
but when I type pdb4amber

[amber18]$ pdb4amber
Traceback (most recent call last):
  File "/users/debarati/amber18/bin/pdb4amber", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/users/debarati/amber18/miniconda/lib/python2.7/site-packages/pkg_resources/__init__.py", line 3126, in <module>
    @_call_aside
  File "/users/debarati/amber18/miniconda/lib/python2.7/site-packages/pkg_resources/__init__.py", line 3110, in _call_aside
    f(*args, **kwargs)
  File "/users/debarati/amber18/miniconda/lib/python2.7/site-packages/pkg_resources/__init__.py", line 3139, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/users/debarati/amber18/miniconda/lib/python2.7/site-packages/pkg_resources/__init__.py", line 581, in _build_master
    ws.require(__requires__)
  File "/users/debarati/amber18/miniconda/lib/python2.7/site-packages/pkg_resources/__init__.py", line 898, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/users/debarati/amber18/miniconda/lib/python2.7/site-packages/pkg_resources/__init__.py", line 784, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'pdb4amber==1.7.dev0' distribution was not found and is required by the application
[amber18]$

**Comment by hainm** [comment]
hi @DDGmichigan, how did you install ambertools?
If installing from source code, have you "source amber.sh" yet? 
Do you see the folder 
`/users/debarati/amber18/lib/python2.7/site-packages/pdb4amber (or similar)`

**Comment by DDGmichigan** [comment]
I had it installed already. I just copied the entire folder of amber18 from /usr/local to my own home directory...

**Comment by DDGmichigan** [comment]
This is the list of contents I found inside /users/debarati/amber18/lib/python2.7/site-packages

[amber18]$ cd lib/python2.7/site-packages/
[site-packages]$ ls
AmberLite-16.0-py2.7.egg-info   mpi4py-2.0.0-py2.7.egg-info                       pdb4amber-1.7.dev0-py2.7.egg         sander-16.0-py2.7.egg-info
AmberTools-17.0-py2.7.egg-info  packmol_memgen                                    pymsmt                               sanderles
easy-install.pth                packmol_memgen-0.9.9a0-py2.7.egg-info             pyMSMT-18.0-py2.7.egg-info           site.py
mdoutanalyzer                   packmol_memgen-1.0.5rc0-py2.7.egg-info            pyMSMT-19.0-py2.7.egg-info           site.pyc
MMPBSA_mods                     ParmEd-3.0.0+57.g74a84d30-py2.7-linux-x86_64.egg  pytraj-2.0.3-py2.7-linux-x86_64.egg
MMPBSA.py-16.0-py2.7.egg-info   ParmEd-3.2.0-py2.7-linux-x86_64.egg               pytraj-2.0.5-py2.7-linux-x86_64.egg
mpi4py                          pdb4amber-1.6.dev0-py2.7.egg                      sander

**Comment by DDGmichigan** [comment]
There are 2 versions of pdb4amber folder : 
pdb4amber-1.6.dev0-py2.7.egg/  and   pdb4amber-1.7.dev0-py2.7.egg/

**Comment by hainm** [comment]
>  I just copied the entire folder of amber18 from /usr/local to my own home directory...

oh, you're not supposed to do this. If you view the content of the `pdb4amber` program like
```
vim `which pdb4amber`
```
you might see that it will look for non-existing folder (since you moved).

**Comment by hainm** [comment]
> (since you moved).

hah, you copied.

**Comment by hainm** [comment]
> pkg_resources.DistributionNotFound: The 'pdb4amber==1.7.dev0' distribution was not found and is required by the application

I guess the 1.7.dev0 is not included in PYTHONPATH yet (due to folder moving?)
It is normally set if you "source amber.sh" (normally work if you don't move your installed folder around).

**Comment by hainm** [comment]
It's worth checking if ` /users/debarati/amber18/lib/python2.7/site-packages` is included in PYTHONPATH (`echo $PYTHONPATH`).

**Comment by DDGmichigan** [comment]
Got it working!

Thank you 👍

**Comment by hainm** [comment]
What did you do?

**Comment by DDGmichigan** [comment]
use source amber.sh
all things worked as it should!

**Comment by hainm** [comment]
thanks @DDGmichigan

**Comment by calgator1317** [comment]
Hi guys! I have the same issue as @DDGmichigan had. I've tried to source it (not sure if I knew or did it properly). I also made sure that  /users/debarati/amber18/lib/python2.7/site-packages is included in PYTHONPATH as @hainm suggested. 
Can you help me to troubleshoot this issue? Thank you

![image](https://user-images.githubusercontent.com/44783102/80865200-bda46d80-8c55-11ea-9b01-8a833d06754d.png)

**Comment by hainm** [comment]
@calgator1317 what does below command give you?
`amber.python -c 'import pdb4amber; print(pdb4amber)'`

**Comment by calgator1317** [comment]
My apology for the late response. I got caught up with doing other tutorials.

**$pdb4amber**                           

Traceback (most recent call last):
  File "/Applications/amber18/bin/pdb4amber", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/Applications/amber18/miniconda/lib/python2.7/site-packages/pkg_resources/__init__.py", line 3241, in <module>
    @_call_aside
  File "/Applications/amber18/miniconda/lib/python2.7/site-packages/pkg_resources/__init__.py", line 3225, in _call_aside
    f(*args, **kwargs)
  File "/Applications/amber18/miniconda/lib/python2.7/site-packages/pkg_resources/__init__.py", line 3254, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/Applications/amber18/miniconda/lib/python2.7/site-packages/pkg_resources/__init__.py", line 583, in _build_master
    ws.require(__requires__)
  File "/Applications/amber18/miniconda/lib/python2.7/site-packages/pkg_resources/__init__.py", line 900, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/Applications/amber18/miniconda/lib/python2.7/site-packages/pkg_resources/__init__.py", line 786, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'pdb4amber==1.7.dev0' distribution was not found and is required by the application

**$amber.python -c 'import pdb4amber; print(pdb4amber)'**

Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named pdb4amber

**Comment by hainm** [comment]
>  I also made sure that /users/debarati/amber18/lib/python2.7/site-packages is included in PYTHONPATH as @hainm suggested.

Hi
I am confused here. This is not your path. It should be `/Applications/amber18/lib/python2.7/site-packages`, right? 

- check if that is in PYTHONPATH
- check if there is pdb4amber folder in that path. If not, there's something wrong with your install.

**Comment by hainm** [comment]
just follow amber manual: `source  /Applications/amber18/amber.sh`

**Comment by calgator1317** [comment]
I don't know how I missed this step in the installation. But thank you so much Hai!

Loc

**Comment by ayesha-69** [comment]
Hi All
I am a having the same issue as above but not sure what should I do. 
I understand the issue or thread is closed. However, if someone could still help out, I will be grateful

I installed the amber on WSL in my computer, no moving files. source is set to  "/mnt/d/amber/amber18/amber.sh"
I tried looking for the python path it says "/usr/bin"
how to add it or where to add it to read the pdb4amber

thanks
